### PR TITLE
Remove JULIA_ENABLE_THREADS feature flag

### PR DESCRIPTION
**Pull request by mbauman**
This flag was essentially not documented, not CI'ed, and broken (it hangs in `jl_task_get_next at /home/mbauman/julia/src/partr.c:475`). Let's strip it out!

### CODE REVIEW

File: src/signals-mach.c [hunk 1]
```diff
@@ -203,12 +194,6 @@ kern_return_t catch_exception_raise(mach_port_t            exception_port,
     HANDLE_MACH_ERROR("thread_get_state", ret);
     uint64_t fault_addr = exc_state.__faultvaddr;
     if (jl_addr_is_safepoint(fault_addr)) {
-#ifdef JULIA_ENABLE_THREADING
```
Comment (smldis) [inline src/signals-mach.c]: just a check: it seems this is the only code in this PR that is actually removing code corresponding to the  `JULIA_ENABLE_THREADING==true` case.

Comment (mbauman) [reply src/signals-mach.c]: Ah good catch.  That explains the macOS failure.

**Comment by mbauman** [comment]
Hm, interesting.  I think we had previously been running analyzegc without the `JULIA_ENABLE_THREADS` flag defined!

**Review by JeffBezanson — APPROVED** [review]

**Comment by JeffBezanson** [comment]
I'd like to merge this. Just checking in with @vtjnash @yuyichao

**Comment by yuyichao** [comment]
Lgtm. Given c11/c++11 I don't think it'll be a big problem supporting threading on a new platform when needed.

**Comment by vtjnash** [comment]
Yep, I think we should start making threads mandatory. Most CPUs now have them. (Even GPUs?)

**Comment by vchuravy** [comment]
Turns out we do have an (experimental) target that currently doesn't support threading.
I had to revert this locally for WASM.

> (Even GPUs?)
Not in the way you are thinking.

**Pull request merged**